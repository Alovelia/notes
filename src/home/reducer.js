import { constants, actions } from 'ducks-helpers';
import handleActions from 'redux-actions/lib/handleActions';
import _ from 'common/helpers';

export const TYPE = constants('home', [
  '~GET',
  //†type
]);
export const ACTION = actions(TYPE);

// The initial state
export const initialState = _.fromJS({
  folders: []
});

export default handleActions({
  [TYPE.GET_SUCCESS]: getSuccess,
  //†handler
}, initialState);

function getSuccess(state, { payload }) {
  return state.set('folders', _.fromJS(payload));
}
//†reducer
