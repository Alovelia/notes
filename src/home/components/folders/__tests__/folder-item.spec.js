import { shallow } from 'enzyme';
import React from 'react';
import noop from 'lodash/noop';
import FolderItem from '../folder-item';
import { ACTION } from '../../../reducer';

describe('<FolderItem />', () => {
  let wrapper;
  let stub;
  beforeAll(() => {
    // stub = sinon.stub(ACTION, 'get').returns({ type: 'mock' });
  });
  afterAll(() => {
    // stub.restore();
  });
  it('when initializing, the expected output is rendered', () => {
    const props = {
      // trigger: noop
      item: {}
    };
    wrapper = shallow(<FolderItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
