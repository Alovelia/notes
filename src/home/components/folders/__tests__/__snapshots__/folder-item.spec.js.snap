// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderItem /> when initializing, the expected output is rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FolderItem
    item={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MediaFigure
          className={null}
        >
          <pure(Folder) />
        </MediaFigure>,
        <MediaBody>
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/undefined"
          />
        </MediaBody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <pure(Folder) />,
          "className": null,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/undefined"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "onlyActiveOnIndex": false,
            "style": Object {},
            "to": "/undefined",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MediaFigure
            className={null}
          >
            <pure(Folder) />
          </MediaFigure>,
          <MediaBody>
            <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/undefined"
            />
          </MediaBody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <pure(Folder) />,
            "className": null,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "function",
          "props": Object {
            "children": <Link
              onlyActiveOnIndex={false}
              style={Object {}}
              to="/undefined"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "onlyActiveOnIndex": false,
              "style": Object {},
              "to": "/undefined",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
